### ISSUE TITLE
Cannot reproduce the result of X-Der reported in paper.

### ISSUE THREAD

**Issue by Taylorfire** [aimagelab/mammoth#21]
Hi!
I try to use the code (without any change)  to reproduce the result of X-Der on Cifar-100 with buffer size of 500. However, I could not reproduce the result 49.93 but only reach the results of 48.34(average), 48.89(max) after 10 experiments. Could you please give any help or explanation?
Besides, I notice that another work of your group LiDER also follow the same benchmark but reports different results on the competitor methods (like icarl and x-Der-RPC) with the results in your paper. 
I plan to follow your work but these two problems make me confused. It will be thankful if you could give any advice or explanation, thanks!

**Comment by mbosc** [comment]
Hello there,

The very slight difference in the results of X-DER might possibly be due to the soft rewrite of the source code we did when publishing the code. It might be that we introduced some very slight difference and the best hyperparameters might no longer have the same exact effect.

Regarding LiDER, there is a major setting difference: buffer size is 64 there and 32 in the original X-DER work. In our experience, batch size has a very large effect on continual leanring experiments (that's why we keep it the same for all competitors in our papers).

**Comment by Taylorfire** [comment]
> Hello there,
> 
> The very slight difference in the results of X-DER might possibly be due to the soft rewrite of the source code we did when publishing the code. It might be that we introduced some very slight difference and the best hyperparameters might no longer have the same exact effect.
> 
> Regarding LiDER, there is a major setting difference: batch size is 64 there and 32 in the original X-DER work. In our experience, batch size has a very large effect on continual leanring experiments (that's why we keep it the same for all competitors in our papers).
> 
> EDIT: batch size, not buffer size, sorry

Thanks！
I have already reproduced a reasonable result on seq-miniimg with X-DER with the hyperparameters you provide. Although there is still a gap on seq-cifar100, I think if it is acceptable to use the result I reproduced with X-DER in new paper as a competitor, that will be a minor problem.
Thanks again for your reply!

**Comment by mbosc** [comment]
Absolutely, if the gap is the one you mentioned above I see no problem at all. Regarding CIFAR-10, we're doing a gridsearch ourselves for X-DER and we'll report the results as soon as we are done!

**Comment by Taylorfire** [comment]
> Absolutely, if the gap is the one you mentioned above I see no problem at all. Regarding CIFAR-10, we're doing a gridsearch ourselves for X-DER and we'll report the results as soon as we are done!

That will be great! Thanks for your time and efforts!

**Comment by loribonna** [comment]
Hi there, not sure if this is still relevant but results on CIFAR-10 are in #34
